packages/jest-haste-map/src/crawlers/node.ts

```diff
@@ -21,6 +22,14 @@ type Result = Array<[/* id */ string, /* mtime */ number, /* size */ number]>;
 
 type Callback = (result: Result) => void;
 
+function hasNativeFindSupport(forceNodeFilesystemAPI: boolean): Promise<void> {
```

Comment:
can this be `Promise<boolean>` instead of relying on rejections? then do `which('find').then(() => true).catch(() => false)`

Also would be nice as an async function.

Reply:
We can probably just do `which.sync('find')` as well, and not have to deal with it being async (would negate my comment about `catch` below)

Reply:
@SimenB first of all thanks for the review. I have some concerns related to the use of blocking operation and i don't have the overall vision of how this can impact jest, but if it's ok to you I can refactor.

Reply:
@SimenB dig into `jest-haste-map` code I found this [comment](https://github.com/facebook/jest/blob/1ed46e71a058c93b529e5e3f9388a800352de21a/packages/jest-haste-map/src/index.ts#L168) and took me out of the worry of the operation being synchronous since this is something expected for this package. Therefore I already did the refactor to use `which.sync` API.

Reply:
since `crawl` itself is async, I moved back the the interface you started with, but I lifted up the promise call out of the `new Promise` constructor, so my earlier worry was solved.